.buildkite/commands/run-unit-tests.sh

```diff
@@ -11,5 +11,26 @@ gem install bundler
 echo "--- :rubygems: Setting up Gems"
 install_gems
 
-echo "--- 🧪 Testing"
+echo "--- 🔬 Testing"
+set +e
 bundle exec fastlane test_without_building name:WordPressUnitTests try_count:3
+TESTS_EXIT_STATUS=$?
+set -e
+
+if [[ "$TESTS_EXIT_STATUS" -ne 0 ]]; then
+  # Keep the (otherwise collapsed) current "Testing" section open in Buildkite logs on error. See https://buildkite.com/docs/pipelines/managing-log-output#collapsing-output
+  echo "^^^ +++"
+  echo "Unit Tests failed!"
+fi
+
+echo "--- 📦 Zipping test results"
+cd build/results/ && zip -rq WordPress.xcresult.zip WordPress.xcresult
+
+echo "--- 🚦 Report Tests Exit Status"
+if [[ $TESTS_EXIT_STATUS -eq 0 ]]; then
+  echo "Unit Tests seems to have passed (exit code 0). All good 👍"
+else
+  echo "The Unit Tests, ran during the '🔬 Testing' step above, have failed."
+  echo "For more details about the failed tests, check the logs under the '🔬 Testing' section and the \`.xcresult\` and test reports in Buildkite artefacts."
+fi
+exit $TESTS_EXIT_STATUS
```

Comment:
Nitpick (the same GitHub is making with that red symbol at the end of the line):

```suggestion
exit $TESTS_EXIT_STATUS

```

The convention of having a new line at the end of the files comes from an old C decision that has been passed down through Unix history:

> A source file that is not empty shall end in a new-line character, which shall not be immediately preceded by a backslash character.
> 
> Since this is a “shall” clause, we must emit a diagnostic message for a violation of this rule.

So, it turns out that, according to POSIX, every text file (including Ruby and JavaScript source files) should end with a `\n`, or “newline” (not “a new line”) character. This acts as the `eol`, or the “end of line” character. It is a line “terminator”. ([Source](https://thoughtbot.com/blog/no-newline-at-end-of-file))

Reply:
I was spoiled by my IDE always adding it. =)
Added in [bbab751](https://github.com/wordpress-mobile/WordPress-iOS/pull/17999/commits/bbab75175d3c595d58526cad4a20a9b5b03676f1)